src/test_instances.cc

```diff
@@ -19,24 +19,21 @@ struct Event {
 static void Notify(void* event) {
   auto& ev = *static_cast<Event*>(event);
 
-  {
-    const std::lock_guard lock(ev.m);
-    ev.ready = true;
-  }
-
+  const std::lock_guard lock(ev.m);
+  ev.ready = true;
   ev.cv.notify_one();
```

Comment:
Otherwise ~Event() in StartTestServer() races against notify_one(). I think this is one of the simpler solutions. Otherwise, maybe use the global scope, return the struct with the thread t1 or pass Event to StartTestServer.